src/platform/unittest/TlsTest.cpp

```diff
@@ -974,6 +974,14 @@ TEST_F(TlsTest, HandshakeFragmented)
     DoHandshake(ServerContext, ClientContext, 200);
 }
 
+TEST_F(TlsTest, HandshakeVeryFragmented)
+{
+    TlsContext ServerContext, ClientContext;
+    ServerContext.InitializeServer(ServerSecConfig, false, 256);
+    ClientContext.InitializeClient(ClientSecConfigNoCertValidation);
+    DoHandshake(ServerContext, ClientContext, 1);
```

Comment:
So this didn't repro originally? Can you try a much bigger TP size (> 1500) on both client and server? My only other guess would be that the difference is related to how QUIC core code buffers until it has a complete message before it calls. We might need to duplicate that logic here to test it.